runtime/common/src/message_router/location_conversion.rs

```diff
@@ -0,0 +1,57 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2022 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+use sp_std::marker::PhantomData;
+use xcm::{latest::MultiLocation, prelude::*};
+use xcm_executor::traits::Convert;
+
+/// The address prefix for dvm address
+const ADDR_PREFIX: &[u8] = b"dvm:";
+
+/// Derive AccountKey20 in MultiLocation to AccountId32
+/// Refer to https://github.com/darwinia-network/darwinia-common/blob/main/frame/support/src/evm.rs#L85
+pub struct AccountKey20Derive<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone> Convert<MultiLocation, AccountId>
+	for AccountKey20Derive<AccountId>
```

Comment:
This is used for converting other key types to Darwinia's AccountId32 format and compatible with H160 converting at the same time right?

Reply:
This is just used for converting [u8;20] (compatible with H160) key type to AccountId32 format.